From 796fb395ee0edb2c340d39d63d2ae332e4bd8e5b Mon Sep 17 00:00:00 2001
From: "Yan, WalterX" <walterx.yan@intel.com>
Date: Mon, 3 Dec 2018 15:22:32 +0800
Subject: [PATCH] Check if mClingWindow is already added before adding.

If not to check if it's added, the previous one will leak
and cannot be removed any more.

Change-Id: Ifd50b57badd8c630aaa1e202695c7c4525b3a30e
Tracked-On: OAM-72403
Signed-off-by: Yan, WalterX <walterx.yan@intel.com>
---
 .../java/com/android/server/policy/ImmersiveModeConfirmation.java     | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/core/java/com/android/server/policy/ImmersiveModeConfirmation.java b/services/core/java/com/android/server/policy/ImmersiveModeConfirmation.java
index 4aa2446..d1df556 100644
--- a/services/core/java/com/android/server/policy/ImmersiveModeConfirmation.java
+++ b/services/core/java/com/android/server/policy/ImmersiveModeConfirmation.java
@@ -354,6 +354,10 @@ public class ImmersiveModeConfirmation {
 
     private void handleShow() {
         if (DEBUG) Slog.d(TAG, "Showing immersive mode confirmation");
+        if (mClingWindow != null) {
+            if (DEBUG) Slog.d(TAG, "already shown");
+            return;
+        }
 
         mClingWindow = new ClingWindowView(mContext, mConfirm);
 
-- 
1.9.1

